src/qt/governancelist.cpp

```diff
@@ -0,0 +1,228 @@
+#include <qt/governancelist.h>
+#include <qt/forms/ui_governancelist.h>
+
+#include <qt/clientmodel.h>
+#include <clientversion.h>
+#include <coins.h>
+#include <qt/guiutil.h>
+#include <netbase.h>
+#include <qt/walletmodel.h>
+#include <governance/governance-object.h>
+
+#include <univalue.h>
+
+#include <QMessageBox>
+#include <QTableWidgetItem>
+#include <QtGui/QClipboard>
+
+
+template <typename T>
+class CGovernanceListWidgetItem : public QTableWidgetItem
+{
+    T itemData;
+
+public:
+    explicit CGovernanceListWidgetItem(const QString& text, const T& data, int type = Type) :
+        QTableWidgetItem(text, type),
+        itemData(data) {}
+
+    bool operator<(const QTableWidgetItem& other) const
+    {
+        return itemData < ((CGovernanceListWidgetItem*)&other)->itemData;
+    }
+};
+
+GovernanceList::GovernanceList(QWidget* parent) :
+    QWidget(parent),
+    ui(new Ui::GovernanceList),
+    clientModel(0),
+    fGovernanceFilterUpdated(true),
+    nTimeGovernanceFilterUpdated(0),
+    nTimeGovernanceUpdated(0),
+    governanceListChanged(true)
+{
+    ui->setupUi(this);
+
+    GUIUtil::setFont({ui->label_count_2, ui->countLabel}, GUIUtil::FontWeight::Bold, 14);
+    GUIUtil::setFont({ui->label_filter_2}, GUIUtil::FontWeight::Normal, 15);
+
+    int columnHashWidth = 80;
+    int columnNameWidth = 220;
+    int columnCreationWidth = 110;
+    int columnStartWidth = 110;
+    int columnEndWidth = 110;
+    int columnAmountWidth = 80;
+    int columnUrlWidth = 200;
+    int columnTypeWidth = 50;
+    int columnActiveWidth = 50;
+
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_HASH, columnHashWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_NAME, columnNameWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_CREATION, columnCreationWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_START, columnStartWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_END, columnEndWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_AMOUNT, columnAmountWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_URL, columnUrlWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_TYPE, columnTypeWidth);
+    ui->tableWidgetGovernances->setColumnWidth(COLUMN_ACTIVE, columnActiveWidth);
+
+    ui->tableWidgetGovernances->setContextMenuPolicy(Qt::CustomContextMenu);
+    ui->tableWidgetGovernances->sortItems(COLUMN_CREATION, Qt::DescendingOrder);
+
+
+    ui->filterLineEdit->setPlaceholderText(tr("Filter by Name"));
+
+    timer = new QTimer(this);
+    connect(timer, SIGNAL(timeout()), this, SLOT(updateGovernanceListScheduled()));
```

Comment:
This should make linter happy again
```suggestion
    connect(timer, &QTimer::timeout, this, &GovernanceList::updateGovernanceListScheduled);
```